release/updateChangeLog.js

```diff
@@ -0,0 +1,104 @@
+const { writeToFile } = require('./util/file')
+const { cleanVersion, cleanVersionIfRC } = require('./util/helpers')
+const parseChangelog = require('changelog-parser')
+const moment = require('moment')
+const semver = require('semver')
+const { debug } = require('@actions/core')
+const { RELEASE_VERSION } = process.env
+
+const updateChangeLog = async () => {
+  let changelogData = await parseChangelog('CHANGELOG.md')
+
+  const alreadyExists = !!changelogData.versions.find((i) => {
+    if (!i.version) {
+      return false
+    }
+    // Only overwrite changelog entries from rc releases
+    return cleanVersionIfRC(i.version) == cleanVersion(RELEASE_VERSION)
+  })
+
+  changelogData.versions = changelogData.versions
+    .map((i) => {
+      const isNextVersion = !!i.title.match('next-version')
+      const date = moment().format('YYYY-MM-DD')
+
+      // For new entry replace next-version (add now one later) otherwise update entry
+      if (
+        (alreadyExists &&
+          cleanVersionIfRC(i.version || '') ===
+            cleanVersion(RELEASE_VERSION)) ||
+        (!alreadyExists && isNextVersion)
+      ) {
+        return {
+          ...i,
+          version: RELEASE_VERSION,
+          date,
+        }
+      }
+
+      return i
+    })
+    .sort((a, b) => {
+      if (a.version === null) {
+        return -1
+      }
+      if (b.version === null) {
+        return 1
+      }
+      return semver.lt(a.version, b.version)
+        ? 1
+        : semver.gt(a.version, b.version)
+        ? -1
+        : 0
+    })
+
+  if (!alreadyExists) {
+    changelogData.versions.unshift({
+      version: 'next-version',
+      title: 'next-version',
+    })
+  }
+
+  debug('Updating changelog')
+  await generateChangeLog(changelogData)
+}
+
+const generateChangeLog = async (data) => {
+  const entries = data.versions
+    .map((i) =>
+      [
+        `## [${i.version || 'next-version'}]${i.date ? ` - ${i.date}` : ''}`,
+        i.body ? `\n\n${i.body}\n\n` : '\n\n',
+      ].join('')
+    )
+    .join('')
+
+  const links = data.versions
+    .map((i, index) => {
+      const isNextVersion = index === 0
+      const isLastEntry = index === data.versions.length - 1
+
+      const previous = isLastEntry ? '0.4.0' : data.versions[index + 1]?.version
```

Comment:
What's this hard coded value here?

Reply:
The earliest version mentioned in our CHANGELOG.md is `0.5.0` and our earliest release tag on github is `0.4.0`. In order to construct the first url on the bottom of our change log (for `0.5.0`) we need `0.4.0` but as it isn't in the changelog, I opted for a simple hard coded value as it won't change anymore.

Reply:
We can assign it to a variable e.g. `const oldestVersion = '0.4.0'`